packages/dds/matrix/src/matrix.ts

```diff
@@ -444,28 +450,46 @@ export class SharedMatrix<T extends Serializable = Serializable>
         }
     }
 
-    public snapshot(): ITree {
-        return {
+    public snapshot(): IChannelSnapshotDetails {
+        // Create a SnapshotSerializer that will be used to serialize any IFluidHandles in this matrix. The
+        // IFluidHandles represents route to any referenced fluid object.
+        // SnapshotSerializer tracks the routes of all handles that it serializes.
+        const serializer = new SnapshotSerializer(this.runtime.IFluidHandleContext);
```

Comment:
Can you please use runtime.channelsRoutingContext instead? It's the same thing today, but it will be different in future and IFluidHandleContext will be gone. Reason - we need to differentiate correctly paths to DDSs vs. objects, now it's same naming space, and thus we can have collisions

Reply:
Updated to use runtime.channelsRoutingContext.